virtual_rainforest/models/abiotic/wind.py

```diff
@@ -0,0 +1,551 @@
+    wind_profile_above = calculate_wind_above_canopy(
+        wind_heights=wind_heights,
+        zero_plane_displacement=zero_plane_displacement,
+        roughness_length_momentum=roughness_length_momentum,
+        diabatic_correction_momentum_above=diabatic_correction_momentum_above,
+        friction_velocity=data["friction_velocity"],
+    )
+
+    wind_profile_canopy = calculate_wind_below_canopy(
+        canopy_node_heights=canopy_node_heights,
+        wind_profile_above=wind_profile_above,
+        wind_attenuation_coefficient=wind_attenuation_coefficient,
+        canopy_height=data["canopy_height"],
+    )
+
+    return wind_profile_above, wind_profile_canopy
+
+
+def calculate_wind_above_canopy(
+    wind_heights: DataArray,
+    zero_plane_displacement: DataArray,
+    roughness_length_momentum: DataArray,
+    diabatic_correction_momentum_above: DataArray,
+    friction_velocity: DataArray,
+) -> DataArray:
+    """Calculate wind profile above canopy.
+
+    Wind profiles above the canopy dictate heat and vapor exchange between the canopy
+    and air above it, and therefore ultimately determine temperature and vapor profiles.
+    We follow the implementation by :cite:t:`Campbell2012` as described in
+    :cite:t:`MACLEAN2021`.
+
+    Args:
+        wind_heights: vector of heights above canopy for which wind speed is calculated,
+            [m]
+        zero_plane_displacement: height above ground within the canopy where the wind
+            profile extrapolates to zero, [m]
+        roughness_length_momentum: roughness length for momentum, [m]
+        diabatic_correction_momentum_above: diabatic correction for momentum above
+            canopy
+        friction_velocity: friction velocity, [m s-1]
+
+    Returns:
+        vertical wind profile above canopy, [m s-1]
+
+    # TODO: find correct fill value
+    """
+
+    return (
+        (friction_velocity / 0.4)
+        * np.log((wind_heights - zero_plane_displacement) / roughness_length_momentum)
+        + diabatic_correction_momentum_above
+    ).fillna(0)
+
+
+def calculate_wind_below_canopy(
+    canopy_node_heights: DataArray,
+    wind_profile_above: DataArray,
+    wind_attenuation_coefficient: DataArray,
+    canopy_height: DataArray,
+) -> DataArray:
+    """Calculate wind profile below canopy.
+
+    Implementation after :cite:t:`MACLEAN2021`.
+    The top of canopy windspeed is taken from the above canopy wind profile, which must
+    be ordered from highest to lowest level. The lowest level is used here as the
+    top-of-canopy wind speed.
+
+    Args:
+        copy_node_heights: heights of canopy nodes, [m]
+        wind_profile_above: wind profile above canopy, [m s-1]
+        wind_attenuation_coefficient: wind attenuation coefficient, dimensionless
+        canopy_height: canopy height, [m]
+
+    Returns:
+        wind profile below canopy, [m s-1]
+
+    # TODO: find correct fill value
+    """
+    topofcanopy_wind_speed = wind_profile_above[:, -1]
```

Comment:
Feel this should be `top_of_canopy_wind_speed` or `wind_speed_canopy_top` if you need it shorter